e_os.h

```diff
@@ -544,6 +544,8 @@ struct servent *getservbyname(const char *name, const char *proto);
 # define CRYPTO_memcmp memcmp
 #endif
 
+#define IF_FAIL_ASSERT(x)     assert(x); if (!(x))
```

Comment:
Another option:

````
#define DECLARE_ASSERT        int ossl_assert_val
#define IF_FAIL_ASSERT(x)     assert(ossl_assert_val = (int)(x)); if (!ossl_assert_val)
````
Which in usage would look like it does now, with the added requirement that you point `DECLARE_ASSERT;` at the start of the function. I could live with that.